packages/apps/shopify-app-remix/src/react/components/RemixPolarisLink.tsx

```diff
@@ -9,6 +9,83 @@ import type {
 export const RemixPolarisLink = React.forwardRef<
   HTMLAnchorElement,
   LinkLikeComponentProps
->((props, ref) => (
-  <Link {...props} to={props.url ?? props.to} ref={ref} />
-)) as LinkLikeComponent;
+>((props, forwardedRef) => {
+  const [element, setElement] = useState<HTMLElement>();
+  const navigate = useNavigate();
+  const navigateRef = useRef(navigate);
+  navigateRef.current = navigate;
+
+  useEffect(() => {
+    async function onClick(event: MouseEvent) {
+      const isAnchorTag = event.target instanceof HTMLAnchorElement;
+      if (!hasAppBridgeCDN() || !isAnchorTag) return;
+
+      const anchor = event.target;
+      const url = parseUrl(anchor.href);
+      if (!url || url.origin !== window.location.origin) return;
+
+      const leaveConfirmation = getLeaveConfirmation();
+      if (!leaveConfirmation) return;
+
+      event.preventDefault();
+      event.stopPropagation();
+
+      await leaveConfirmation().catch(noop);
+
+      const {pathname, search, hash} = url;
+      navigateRef.current(`${pathname}${search}${hash}`);
+    }
+    element?.addEventListener('click', onClick);
+    return () => {
+      element?.removeEventListener('click', onClick);
+    };
+  }, [element]);
+
+  return (
+    <Link
+      {...props}
+      to={props.url ?? props.to}
+      ref={(anchor) => {
+        setElement(anchor || undefined);
+        if (forwardedRef) {
+          if (typeof forwardedRef === 'function') {
+            forwardedRef(anchor);
+          } else {
+            forwardedRef.current = anchor;
+          }
+        }
+      }}
+    />
+  );
+}) as LinkLikeComponent;
+
+function parseUrl(url: string | undefined | null) {
```

Comment:
Total nit, but does `undefined` vs `null` matter here? It seems like we could just use `undefined` all around?